build/targets/KoreBuild.DefaultBuildSettings.props

```diff
@@ -1,4 +1,16 @@
 <Project>
+  <!-- pickup build number from CI if available -->
+  <PropertyGroup Condition=" '$(BuildNumber)' == '' ">
+    <!--
+      TeamCity=https://confluence.jetbrains.com/display/TCD10/Predefined+Build+Parameters
+      AppVeyor=https://www.appveyor.com/docs/environment-variables/
+      Travis=https://docs.travis-ci.com/user/environment-variables/
+    -->
+    <BuildNumber Condition=" '$(BUILD_NUMBER)' != '' ">$(BUILD_NUMBER)</BuildNumber>
```

Comment:
Could you make sure that we're setting `BuildCounter` from a previous dependency everywhere we need to and not setting `BuildNumber` to `dep.PreviousConfig.BuildNumber`?